csharp/ql/src/semmle/code/csharp/commons/Assertions.qll

```diff
@@ -81,18 +100,9 @@ class Assertion extends MethodCall {
    */
   pragma[nomagic]
   predicate strictlyDominates(BasicBlock bb) {
-    this.getAControlFlowNode().getBasicBlock().immediatelyDominates(bb)
+    this.strictlyDominatesSplit(bb)
     or
-    if bb instanceof JoinBlock then
-      this.isPossiblyDominatedJoinBlock(bb) and
-      forall(BasicBlock pred |
-        pred = this.getAPossiblyDominatedPredecessor(bb) |
-        this.strictlyDominates(pred)
-        or
-        this.getAControlFlowNode().getBasicBlock() = pred
-      )
-    else
-      this.strictlyDominates(bb.getAPredecessor())
+    this.getAControlFlowNode().getBasicBlock().strictlyDominates(bb)
```

Comment:
This does make sense - my only question is whether `strictlyDominates` belongs in this class - as it seems to be a very general predicate that isn't necessarily tied to `class Assertion`. Perhaps it should be reversed to `bb.isDominatedBy(this)` and implemented there?

Reply:
The problem with that approach is that there is then no restriction on the control flow elements to calculate dominance for, and the result will simply be too big. By adding it here, we restrict the calculation to assertions, which is a much smaller subset.